airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -336,28 +351,64 @@ private ConnectorJobOutput getCheckResponse(final CheckConnectionInput checkInpu
     return runMandatoryActivityWithOutput(checkActivity::runWithJobOutput, checkInput);
   }
 
-  private SyncCheckConnectionFailure checkConnections(final GenerateInputActivity.GeneratedJobInput jobInputs) {
-    final JobRunConfig jobRunConfig = jobInputs.getJobRunConfig();
+  private SyncJobCheckConnectionInputs getCheckConnectionInputFromSync(final GenerateInputActivity.GeneratedJobInput jobInputs) {
     final StandardSyncInput syncInput = jobInputs.getSyncInput();
     final JsonNode sourceConfig = syncInput.getSourceConfiguration();
     final JsonNode destinationConfig = syncInput.getDestinationConfiguration();
     final IntegrationLauncherConfig sourceLauncherConfig = jobInputs.getSourceLauncherConfig();
     final IntegrationLauncherConfig destinationLauncherConfig = jobInputs.getDestinationLauncherConfig();
-    final SyncCheckConnectionFailure checkFailure = new SyncCheckConnectionFailure(jobRunConfig);
 
     final StandardCheckConnectionInput standardCheckInputSource = new StandardCheckConnectionInput()
         .withActorType(ActorType.SOURCE)
         .withActorId(syncInput.getSourceId())
         .withConnectionConfiguration(sourceConfig);
-    final CheckConnectionInput checkSourceInput = new CheckConnectionInput(jobRunConfig, sourceLauncherConfig, standardCheckInputSource);
+
+    final StandardCheckConnectionInput standardCheckInputDestination = new StandardCheckConnectionInput()
+        .withActorType(ActorType.DESTINATION)
+        .withActorId(syncInput.getDestinationId())
+        .withConnectionConfiguration(destinationConfig);
+
+    return new SyncJobCheckConnectionInputs(
+        sourceLauncherConfig,
+        destinationLauncherConfig,
+        standardCheckInputSource,
+        standardCheckInputDestination);
+  }
+
+  private SyncCheckConnectionFailure checkConnections(final JobRunConfig jobRunConfig,
+                                                      @Nullable final GenerateInputActivity.GeneratedJobInput jobInputs) {
+    final SyncCheckConnectionFailure checkFailure = new SyncCheckConnectionFailure(jobRunConfig);
 
     final JobCheckFailureInput jobStateInput =
         new JobCheckFailureInput(Long.parseLong(jobRunConfig.getJobId()), jobRunConfig.getAttemptId().intValue(), connectionId);
     final boolean isLastJobOrAttemptFailure =
         runMandatoryActivityWithOutput(jobCreationAndStatusUpdateActivity::isLastJobOrAttemptFailure, jobStateInput);
-    if (isResetJob(sourceLauncherConfig) || checkFailure.isFailed() || !isLastJobOrAttemptFailure) {
+
+    if (!isLastJobOrAttemptFailure) {
+      log.info("SOURCE CHECK: Skipped, last attempt was not a failure");
+      log.info("DESTINATION CHECK: Skipped, last attempt was not a failure");
+      return checkFailure;
+    }
```

Comment:
I think I'm confused by this conditional - if we are skipping CHECKS because the previous job/attempt didn't fail, why would we return a failure?  Or is checkFailure initialized with `isFailed(false)`

Reply:
This was already the case before, I just moved this higher up to be clearer / more intentional about it. 

Yes, a `SyncCheckConnectionFailure` is always returned by this method. When an actual failure happens, `checkFailure.setFailureOrigin` and `checkFailure.setFailureOutput` are called. The `SyncCheckConnectionFailure` class has a method `.isFailed()` that actually returns whether it failed or not if these are set.

Reply:
Ok, this was definitely confusing :D 

Renamed SyncCheckConnectionFailure to SyncCheckConnectionResult to be more clear that it doesn't _always_ represent a failure.